daemon/api_sideload_n_try.go

```diff
@@ -43,14 +45,33 @@ import (
 
 const maxReadBuflen = 1024 * 1024
 
+type Form struct {
+	Value      map[string][]string
+	FileHeader map[string][]*FileHeader
```

Comment:
Since we're uploading a snap, unless this would interfere with your work on snap bundles, perhaps we do not need a generic form here, but something like:
```
type SnapUploadForm struct {
    Options map[string]string
    TempSnapFile *FileHeader
}
```

Reply:
I left readForm as devoid of "snap" related things as possible on purpose so `sideloadOrTry` can then make its snap logic checks. I think it'll make it easier to extend as well, since this struct and the deserialization would have to change when adding multiple snap files. However, we could add a helper that iterates on the generic form and gets/checks what it needs. That would extract the "snap" related checks from the `sideloadOrTry` function while still making it easy to extend the logic since only the helper would change, not any data structures or deserialisation (I think).